typescript/radix-sdk/src/modules/tx.ts

```diff
@@ -0,0 +1,519 @@
+      from_address: this.account.address,
+      mailbox,
+      name,
+      symbol,
+      description,
+      divisibility,
+    });
+
+    const intentHashTransactionId =
+      await this.signAndBroadcast(transactionManifest);
+
+    return this.getNewComponent(intentHashTransactionId);
+  }
+
+  public async setTokenOwner({
+    token,
+    new_owner,
+  }: {
+    token: string;
+    new_owner: string;
+  }) {
+    const transactionManifest = await this.populate.setTokenOwner({
+      from_address: this.account.address,
+      token,
+      new_owner,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async setTokenIsm({ token, ism }: { token: string; ism: string }) {
+    const transactionManifest = await this.populate.setTokenIsm({
+      from_address: this.account.address,
+      token,
+      ism,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async enrollRemoteRouter({
+    token,
+    receiver_domain,
+    receiver_address,
+    gas,
+  }: {
+    token: string;
+    receiver_domain: number;
+    receiver_address: string;
+    gas: string;
+  }) {
+    const transactionManifest = await this.populate.enrollRemoteRouter({
+      from_address: this.account.address,
+      token,
+      receiver_domain,
+      receiver_address,
+      gas,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async unrollRemoteRouter({
+    token,
+    receiver_domain,
+  }: {
+    token: string;
+    receiver_domain: number;
+  }) {
+    const transactionManifest = await this.populate.unrollRemoteRouter({
+      from_address: this.account.address,
+      token,
+      receiver_domain,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async remoteTransfer({
+    token,
+    destination_domain,
+    recipient,
+    amount,
+    custom_hook_id,
+    gas_limit,
+    custom_hook_metadata,
+    max_fee,
+  }: {
+    token: string;
+    destination_domain: number;
+    recipient: string;
+    amount: string;
+    custom_hook_id: string;
+    gas_limit: string;
+    custom_hook_metadata: string;
+    max_fee: { denom: string; amount: string };
+  }) {
+    const transactionManifest = await this.populate.remoteTransfer({
+      from_address: this.account.address,
+      token,
+      destination_domain,
+      recipient,
+      amount,
+      custom_hook_id,
+      gas_limit,
+      custom_hook_metadata,
+      max_fee,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async signAndBroadcast(
+    manifest: TransactionManifest,
+  ): Promise<TransactionHash> {
+    const constructionMetadata =
+      await this.gateway.transaction.innerClient.transactionConstruction();
+
+    const transactionHeader: TransactionHeader = {
+      networkId: this.networkId,
+      startEpochInclusive: constructionMetadata.ledger_state.epoch,
+      endEpochExclusive: constructionMetadata.ledger_state.epoch + 2,
```

Comment:
Is this similar to Solana's latest block height where if we reach this height then the transaction expires? if it is should we add some kind of retry logic that automatically updates this value and after X number of retries the transaction submission fails?